common/src/main/java/com/microsoft/identity/common/internal/commands/GenerateShrCommand.java

```diff
@@ -0,0 +1,111 @@
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.UiRequiredException;
+import com.microsoft.identity.common.internal.commands.parameters.GenerateShrCommandParameters;
+import com.microsoft.identity.common.internal.controllers.BaseController;
+import com.microsoft.identity.common.internal.result.GenerateShrResult;
+
+import java.util.List;
+
+import static com.microsoft.identity.common.exception.ErrorStrings.NO_ACCOUNT_FOUND;
+
+/**
+ * Command class to perform generation of AT-less SHRs on behalf of a user.
+ */
+public class GenerateShrCommand extends BaseCommand<GenerateShrResult> {
+
+    private static final String TAG = GenerateShrCommand.class.getSimpleName();
+
+    /**
+     * Constructs a new GenerateShrCommand.
+     *
+     * @param parameters  The command's input parameters.
+     * @param controllers The controllers on which to run this command.
+     * @param callback    The command to notify once execution has completed.
+     * @param publicApiId The public API ID of this command.
+     */
+    public GenerateShrCommand(@NonNull final GenerateShrCommandParameters parameters,
+                              @NonNull final List<BaseController> controllers,
+                              @NonNull final CommandCallback<GenerateShrResult, BaseException> callback,
+                              @NonNull final String publicApiId) {
+        super(parameters, controllers, callback, publicApiId);
+    }
+
+    @Override
+    public GenerateShrResult execute() throws Exception {
+        final String methodName = ":execute";
+
+        GenerateShrResult result = null;
+        final GenerateShrCommandParameters parameters = (GenerateShrCommandParameters) getParameters();
+
+        // Iterate over our controllers, to service the request either locally or via the broker...
+        // if the local (embedded) cache contains tokens for the supplied user, we will sign using
+        // the embedded PoP keys. If no local user-state exists, the broker will be delegated to
+        // where the same check is performed.
+        BaseController controller;
+        for (int ii = 0; ii < getControllers().size(); ii++) {
+            controller = getControllers().get(ii);
+
+            com.microsoft.identity.common.internal.logging.Logger.verbose(
+                    TAG + methodName,
+                    "Executing with controller: "
+                            + controller.getClass().getSimpleName()
+            );
+
+            result = controller.generateSignedHttpRequest(parameters);
+
+            if (null != result.getErrorCode()) {
+                final String errorCode = result.getErrorCode();
+                final String errorMessage = result.getErrorMessage();
+
+                // To support a shared communication model between the local flow and the
```

Comment:
what do we mean by shared communication model? are these controllers talking to each other?

Reply:
No no -- this refers to the fact that errors from the various controllers (local or broker) aren't thrown, but are instead returned as properties of the `GenerateShrResult`. So even though we _could_ use `throw` locally, we don't so that the local controller behaves like the broker (where we cannot `throw` Exceptions across process boundaries) I'll clarify this comment

Reply:
Update this comment to change `shared` -> `consistent`